doc/sphinx/language/gallina-specification-language.rst

```diff
@@ -614,148 +619,101 @@ syntax: :n:`let fix @ident @binders := @term in` stands for
-:math:`n=1`, the “``for`` :token:`ident`:math:`_i`” clause is omitted.
+:math:`n=1`, the “``for`` :n:`@ident`:math:`_i`” clause is omitted.
 
 .. _vernacular:
 
 The Vernacular
 ==============
 
-.. insertgramXX vernac ident_opt2
-
-.. productionlist:: coq
-   decorated-sentence : [ `decoration` … `decoration` ] `sentence`
-   sentence           : `assumption`
-                      : `definition`
-                      : `inductive`
-                      : `fixpoint`
-                      : `assertion` `proof`
-   assumption         : `assumption_keyword` `assums`.
-   assumption_keyword : Axiom | Conjecture
-                      : Parameter | Parameters
-                      : Variable | Variables
-                      : Hypothesis | Hypotheses
-   assums             : `ident` … `ident` : `term`
-                      : ( `ident` … `ident` : `term` ) … ( `ident` … `ident` : `term` )
-   definition         : [Local] Definition `ident` [`binders`] [: `term`] := `term` .
-                      : Let `ident` [`binders`] [: `term`] := `term` .
-   binders            : binders binder
-                      : binder
-   inductive          : Inductive `ind_body` with … with `ind_body` .
-                      : CoInductive `ind_body` with … with `ind_body` .
-   ind_body           : `ident` [`binders`] : `term` :=
-                      : [[|] `ident` [`binders`] [:`term`] | … | `ident` [`binders`] [:`term`]]
-   fixpoint           : Fixpoint `fix_body` with … with `fix_body` .
-                      : CoFixpoint `fix_body` with … with `fix_body` .
-   assertion          : `assertion_keyword` `ident` [`binders`] : `term` .
-   assertion_keyword  : Theorem | Lemma
-                      : Remark | Fact
-                      : Corollary | Property | Proposition
-                      : Definition | Example
-   proof              : Proof . … Qed .
-                      : Proof . … Defined .
-                      : Proof . … Admitted .
-   decoration : #[ `attributes` ]
-   attributes : [`attribute`, … , `attribute`]
-   attribute :  `ident`
-             : `ident` = `string`
-             : `ident` ( `attributes` )
-
-.. todo:: This use of … in this grammar is inconsistent
-          What about removing the proof part of this grammar from this chapter
-          and putting it somewhere where top-level tactics can be described as well?
-          See also #7583.
+.. insertprodn vernacular vernacular
+
+.. prodn::
+   vernacular ::= {* {? @all_attrs } {| @command | @ltac_expr } . }
 
-This grammar describes *The Vernacular* which is the language of
-commands of Gallina. A sentence of the vernacular language, like in
-many natural languages, begins with a capital letter and ends with a
-dot.
+The top-level input to |Coq| is a series of commands and tactics, each terminated with a period
+and optionally decorated with :ref:`gallina-attributes`.  :n:`@ltac_expr` syntax supports both simple
```

Comment:
Side-remark: we have a problem with the place `ltac_expr` links to (in the Ltac2 chapter). The grammar chunks extending such tokens in the Ltac2 chapter should probably be converted to use `prodn` with `+=` like in the SSReflect chapter.

Reply:
Hum, why mark this as resolved?

Reply:
This is now its own issue.